node/builder.go

```diff
@@ -0,0 +1,364 @@
+package node
+
+// Review: is a builder pattern with dot-chaining idiomatic in go?
```

Comment:
It's not nearly as common as in other languages, but I am also implementing one for the node builder for tests (possibly I should rename or we can converge).

The [functional options pattern](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis) is often used for similar goals. I don't like it as much for the mutability, but it is canonical. Either is fine here.

Reply:
I will read it up, in any case, it was more to decide what to do with a follow up PR. I will clean up the review and TODO and we can deal with it afterwards.